Google.Api.Generator/Generation/MetadataGenerator.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Google.Gapic.Metadata;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+
+namespace Google.Api.Generator.Generation
+{
+    internal static class MetadataGenerator
+    {
+        private const string GapicMetadataSchemaVersion = "1.0";
+        private const string GapicMetadataFileDescription = "This file maps proto services/RPCs to the corresponding library clients/methods";
+        private const string GapicMetadataLanguage = "csharp";
+
+        private const string TransportKeyGrpc = "grpc";
+
+        internal static string GenerateGapicMetadataJson(List<ServiceDetails> allServiceDetails)
+        {
+            
+            var gapicMetadata = new GapicMetadata
+            {
+                Schema = GapicMetadataSchemaVersion,
+                Comment = GapicMetadataFileDescription,
+                Language = GapicMetadataLanguage,
+                ProtoPackage = allServiceDetails.First().Package,
+                LibraryPackage = allServiceDetails.First().Namespace
+            };
+
+            foreach (var serviceDetails in allServiceDetails)
+            {
+                gapicMetadata.Services[serviceDetails.ServiceName] = ServiceForTransportMetadata(serviceDetails);
+            }
+
+            var gapicMetadataJson = JsonSerializer.Serialize(
+                gapicMetadata,
+                new JsonSerializerOptions
+                {
+                    WriteIndented = true,
+                    PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+                    DictionaryKeyPolicy = JsonNamingPolicy.CamelCase
+                });
+
+            return gapicMetadataJson + "\n"; //trailing file newline to please github
+        }
+
+        private static GapicMetadata.Types.ServiceForTransport ServiceForTransportMetadata(ServiceDetails serviceDetails)
+        {

```

Comment:
This can be implemented as an expression-bodied member:

```csharp
public static ServiceForTransport ServiceForTransportMetadata(ServiceDetails serviceDetails) =>
    new ServiceForTransport
    {
        Clients = { [TransportKeyGrpc] = ServiceAsClientMetadata(serviceDetails) }
    };
```

Reply:
At that point, you might even just want to inline the method into its one caller.

Reply:
`Clients` has no setter, I tried to do it the pretty way

Reply:
now that I know about the `={}` on proto collection classes, I was able to make this whole class much prettier (it will be even prettier when target-typed object creation hits.

Reply:
👍